clang/lib/DPCT/DPCT.cpp

```diff
@@ -614,6 +615,34 @@ int runDPCT(int argc, const char **argv) {
     dpctExit(MigrationSucceeded);
   }
 
+#ifndef _WIN32
+  if (InterceptBuildCommand) {
+    SmallString<512> PathToInterceptBuildBinary(DpctInstallPath);
+    llvm::sys::path::append(PathToInterceptBuildBinary, "bin",
+                            "intercept-build");
+    if (!llvm::sys::fs::exists(PathToInterceptBuildBinary)) {
+      DpctLog() << "Error: intercept-build tool not found"
+                << "\n";
+      ShowStatus(MigrationErrorInvalidInstallPath);
+      dpctExit(MigrationErrorInvalidInstallPath);
+    }
+    std::string InterceptBuildSystemCall(PathToInterceptBuildBinary.str());
+    for (int argumentIndex = 2; argumentIndex < argc; argumentIndex++) {
+      InterceptBuildSystemCall.append(" ");
+      InterceptBuildSystemCall.append(std::string(argv[argumentIndex]));
+    }
+    int processExitCode = system(InterceptBuildSystemCall.c_str());
+    if (processExitCode) {
+      DpctLog() << "Error: intercept-build process call not successful"
+                << "\n";
+      ShowStatus(MigrationError);
```

Comment:
we may define a new error code for intercept-build.

Reply:
Could you suggest an error code name for this please?

Do you think `ProcessError` / `SubcommandError` / `SubCommandExecutionFailure` is suitable?

Reply:
Defined new Error code as `InterceptBuildSuccess` / `InterceptBuildError`. Done.